Source/WebCore/rendering/MarkedText.cpp

```diff
@@ -113,19 +113,39 @@ Vector<MarkedText> MarkedText::collectForHighlights(const RenderText& renderer,
         auto& parentRenderer = *renderer.parent();
         auto& parentStyle = parentRenderer.style();
         if (auto highlightRegister = renderer.document().highlightRegisterIfExists()) {
-            for (auto& highlight : highlightRegister->map()) {
-                auto renderStyle = parentRenderer.getUncachedPseudoStyle({ PseudoId::Highlight, highlight.key }, &parentStyle);
+            for (auto& highlightName : highlightRegister->orderedHighlightNames()) {
+                auto renderStyle = parentRenderer.getUncachedPseudoStyle({ PseudoId::Highlight, highlightName }, &parentStyle);
                 if (!renderStyle)
                     continue;
                 if (renderStyle->textDecorationsInEffect().isEmpty() && phase == PaintPhase::Decoration)
                     continue;
-                for (auto& rangeData : highlight.value->rangesData()) {
+                for (auto& rangeData : highlightRegister->map().get(highlightName)->rangesData()) {
                     if (!highlightData.setRenderRange(rangeData))
                         continue;
 
                     auto [highlightStart, highlightEnd] = highlightData.rangeForTextBox(renderer, selectableRange);
-                    if (highlightStart < highlightEnd)
-                        markedTexts.append({ highlightStart, highlightEnd, MarkedText::Type::Highlight, nullptr, highlight.key });
+
+                    if (highlightStart < highlightEnd) {
+                        int currentPriority = highlightRegister->map().get(highlightName)->priority();
+                        // If we can just append it to the end, do that instead.
+                        if (markedTexts.isEmpty()
+                            || markedTexts.last().priority <= currentPriority) {
+                            markedTexts.append({ highlightStart, highlightEnd, MarkedText::Type::Highlight, nullptr, highlightName, currentPriority });
+                        } else {
+                            // Find the first correct place to insert highlight.
+                            bool inserted = false;
```

Comment:
Boolean variables need a prefix as per WebKit coding style. Should be something like `wasInserted`.